keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,532 @@
+
+###### Test Recommendation
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps  
+
+  - Create a ReplicaSet, in PodTemplate, without specifying `hostIPC`, `hostNetwork`, `hostPID`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - For property in `hostIPC`, `hostNetwork`, `hostPID`
+    - Create a ReplicaSet, in PodTemplate, specifying the property as `true`
+    - Poll the ReplicaSet, should got `ReplicaFailure` condition from `status`, and no Pod is created
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Container Capabilities
+
+  *Conformance Profile: Restricted*
+
+  Containers in Pod created without explicit RBAC associated with the Pod service account
+  (`default` or explicitly specified in `ServiceAccountName`),
+  should have all capabilities dropped.
+
+###### Test Recommendation
+
+  This test requires a container image with the programs to verify Linux capabilities.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet, in PodTemplate/SecurityContext, add capabilities for test in `capabilities`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Check the output of the Pod (can be log) and make sure no capabilities is added
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Limited Volume Types
+
+  *Conformance Profile: Restricted*
+
+  Pod creation with volume types outside of
+  `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  should fail if there's no explicit RBAC associated with the Pod service account
+  (`default` or explicitly specified in `ServiceAccountName`).
+
+###### Test Recommendation
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet, with volumes of types of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Create a ReplicaSet, with a volume with type none of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet, should got `ReplicaFailure` condition from `status`, and no Pod is created
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Run as root
+
+  *Conformance Profile: Isolated*
+
+  Containers in Pod created without explicit RBAC associated with the Pod service account,
+  should run as non-root user and group.
```

Comment:
> and group.

Run as group is an alpha feature.

Reply:
So, shall we keep it out for now? We can add back later when the feature is in beta or GA.